ocs_ci/ocs/ui/page_objects/buckets_tab.py

```diff
@@ -1,18 +1,146 @@
 import json
-
+import logging
+import uuid
 import requests
+import random
+import string
+import os
+import time
+
 from selenium.webdriver.common.by import By
+from selenium.common.exceptions import NoSuchElementException
+
 from ocs_ci.ocs.ocp import get_ocp_url
 from ocs_ci.ocs import exceptions
 from ocs_ci.ocs.ui.page_objects.confirm_dialog import ConfirmDialog
-from ocs_ci.ocs.ui.page_objects.object_storage import ObjectStorage, logger
+from ocs_ci.ocs.ui.page_objects.object_storage import ObjectStorage
 from ocs_ci.utility import version
 
+logger = logging.getLogger(__name__)
+
 
 class BucketsTab(ObjectStorage, ConfirmDialog):
+    """
+    A class representation for abstraction of Buckets tab related OpenShift UI actions
+    """
+
     def __init__(self):
         super().__init__()
 
+    # Methods can directly access locators via self.bucket_tab, self.generic_locators etc.
+    # No need to explicitly import or assign them
+
+    def generate_folder_with_file(self):
+        # TODO: move to utils
+        """
+        Generates a random folder with a random text file inside it in /tmp folder
+
+        Returns:
+            str: Full path to the generated folder
+        """
+        folder_name = "".join(
+            random.choices(string.ascii_lowercase + string.digits, k=8)
+        )
+        folder_path = os.path.join("/tmp", folder_name)
+        os.makedirs(folder_path, exist_ok=True)
+
+        filename = (
+            "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
+            + ".txt"
+        )
+        filepath = os.path.join(folder_path, filename)
+
+        content = "".join(
+            random.choices(string.ascii_letters + string.digits + " \n", k=100)
+        )
+
+        with open(filepath, "w") as f:
+            f.write(content)
+
+        return folder_path
+
+    def create_bucket_ui(self, method):
+
+        self.do_click(self.bucket_tab["create_bucket_button"])
+        if method == "obc":
+            self.create_bucket_via_obc()
+        elif method == "s3":
+            self.create_bucket_via_s3()
+        else:
+            raise ValueError(f"Invalid method: {method}")
+
+    def create_bucket_via_obc(self):
+
+        self.do_click(self.bucket_tab["create_bucket_button_obc"])
+        self.do_click(self.bucket_tab["storage_class_dropdown"])
+        self.do_click(self.bucket_tab["storage_class_noobaa_option"])
+        self.do_click(self.bucket_tab["submit_button_obc"])
```

Comment:
please make this function return Page Object. It should return the instance of the web-page where user left after clicking `submit_button_obc`